server/src/test/java/org/elasticsearch/index/IndexServiceTests.java

```diff
@@ -136,7 +136,6 @@ public void testRefreshTaskIsUpdated() throws IOException {
         assertNotSame(refreshTask, indexService.getRefreshTask());
         assertTrue(refreshTask.isClosed());
         assertFalse(refreshTask.isScheduled());
-        assertFalse(indexService.getRefreshTask().mustReschedule());
```

Comment:
Because checking whether `close()` has been called is now outside of the control of the derived class.  `mustReschedule()` on the derived class just checks extra conditions over and above whether the task has been closed.  (`mustReschedule()` is protected - it's not meant to be called by external users.)  If `AbstractAsyncTaskTests` is doing its job properly then `assertTrue(refreshTask.isClosed());` on line 137 should be enough in this file to confirm that the task will never be scheduled again.

Reply:
k